File: Lua/pe015.lua
--[[
pe015.lua
Starting in the top left corner of a 22 grid, and only being able to move to the right
and down, there are exactly 6 routes to the bottom right corner.

How many such routes are there through a 2020 grid?
--]]

tab={}

dimH=20
dimV=20

for i=0,dimV,1 do
  local a={}
  for j=0,dimH,1 do
    if i==0 then
    a[j]=1
    elseif j==0 then
      a[j]=1
    else
      a[j]=a[j-1]+tab[i-1][j]
    end
  end
  tab[i]=a
end

print(tab[dimV][dimH])

File: Python/PE015.py
##Problem 15
##
##Starting in the top left corner of a 2×2 grid, and only being able to move to the right and down, there are exactly 6 routes to the bottom right corner.
##
##How many such routes are there through a 20×20 grid?


def fact(n):
    p = 1
    for i in range(2, n+1):
        p *= i

    return p


def C_n_k(n, k):
    return fact(n)//(fact(n-k)*fact(k))

print(C_n_k(40,20))        
